content/docs/user-guide/build/generated-source.md

```diff
@@ -0,0 +1,145 @@
+{{< /important >}}
+
+### Data input files
+
+A data input filename can refer to one file explicitly or many files that match a pattern.  If you list a filename explicitly, then it must have a proper path that's absolute or relative to the `CMakeLists.txt` file. If a filename contains a pattern, then the expansion finds all data input files that match the filename.
+
+<!-- TODO: Verify - can explicit filenames be absolute or relative to the CMakeLists.txt file?-->
+
+The input filename supports the following matching operators:
+  * `*` - Sequence of characters any length.
+  * `?` - Single-character sequence.
+  * `[<sequence>]` - Matches any characters in `<sequence>`.
+  * `[!<sequence>]` - Matches any characters _not_ in `<sequence>`.
+
+For information about authoring data input files, refer to [Authoring Jinja templates and data inputs](/docs/user-guide/programming/autogen/).
+
+### Jinja Templates
+
+A template filename must be an explicit path to a single `.jinja` file. The path can be absolute or relative to the `CMakeLists.txt` file. 
+
+For information about authoring Jinja templates, refer to [Authoring Jinja templates and data inputs](/docs/user-guide/programming/autogen/).
+
+
+### Output files
+
+When AzAutoGen feeds data input files into the Jinja2 templates, it generates output files. You must specify the names of the output files you want to generate in the autogen rules. You can use the following *special values* when defining the output filenames:
+
+* `$path`: The path to the final output destination.
+* `$fileprefix`: The name of the current file up to the first `.` token. This is equivalent to running the UNIX shell command `basename ${file%%.*}`.
+* `$file`: The full name of the input file. This is equivalent to running the UNIX shell command `basename $file`.
+
+{{< important >}}
+When generating multiple output files, the output name should _always_ begin with `$path`. Otherwise, AzAutoGen may place output files in an incorrect directory.
+{{< /important >}}
+
+### Example
+
+The following example shows how autogen rules are formatted.
+
+```
+AUTOGEN_RULES
+    # Test wildcarded expansion for xml
+    *.AutoEnum.xml,AutoEnum_Header.jinja,$path/$fileprefix.AutoEnum.h
+    *.AutoEnum.xml,AutoEnum_Source.jinja,$path/$fileprefix.AutoEnum.cpp
+    
+    # Test wildcarded expansion for json
+    *.AutoStruct.json,AutoStruct_Header.jinja,$path/$fileprefix.AutoStruct.h
+    
+    # Test explicit data input file
+    AzCore/AutoGen/WeaponTypes.AutoStruct.json,AutoStruct_Header.jinja,$path/$fileprefix2.AutoStruct.h
+    
+    # Test globbing data input files into a single output file
+    *.AutoEnum.xml,AutoEnumRegistry_Header.jinja,$path/AutoEnumRegistry.h
+```
+
+## Integrating with any target
+
+Most of the time, you want to run AzAutoGen by passing autogen rules into `ly_add_target()`, as outlined in the [Integrating with an O3DE build target](#integrating-with-an-o3de-build-target) section earlier. 
```

Comment:
The `#integrating-with-an-o3de-build-target` reference doesn't appear to be jumping to that section in the deployment preview.